src/Core/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -0,0 +1,354 @@
+        }
+
+        if (string.IsNullOrWhiteSpace(organization.GatewayCustomerId))
+        {
+            throw new BadRequestException("No payment method found.");
+        }
+
+        if (string.IsNullOrWhiteSpace(organization.GatewaySubscriptionId))
+        {
+            throw new BadRequestException("No subscription found.");
+        }
+
+        var plan = StaticStore.SecretManagerPlans.FirstOrDefault(p => p.Type == organization.PlanType);
+
+        if (plan == null)
+        {
+            throw new BadRequestException("Existing plan not found.");
+        }
+
+        if (!plan.HasAdditionalServiceAccountOption)
+        {
+            throw new BadRequestException("Plan does not allow additional Service Account.");
+        }
+
+        var newServiceAccountsTotal = organization.SmServiceAccounts.HasValue
+            ? organization.SmServiceAccounts.Value + serviceAccountAdjustment
+            : serviceAccountAdjustment;
+
+        if (plan.BaseServiceAccount > newServiceAccountsTotal)
+        {
+            throw new BadRequestException($"Plan has a minimum of {plan.BaseServiceAccount} Service Account.");
+        }
+
+        if (newServiceAccountsTotal <= 0)
+        {
+            throw new BadRequestException("You must have at least 1 Service Account.");
+        }
+
+        var additionalServiceAccounts = newServiceAccountsTotal - plan.BaseServiceAccount;
+
+        if (plan.MaxAdditionalServiceAccount.HasValue && additionalServiceAccounts > plan.MaxAdditionalServiceAccount.Value)
+        {
+            throw new BadRequestException($"Organization plan allows a maximum of " +
+                                          $"{plan.MaxAdditionalServiceAccount.Value} additional Service Account.");
+        }
+
+        if (!organization.SmServiceAccounts.HasValue || organization.SmServiceAccounts.Value > newServiceAccountsTotal)
+        {
+            var occupiedServiceAccounts = await _serviceAccountRepository.GetServiceAccountCountByOrganizationIdAsync(organization.Id);
+            if (occupiedServiceAccounts > newServiceAccountsTotal)
+            {
+                throw new BadRequestException($"Your organization currently has {occupiedServiceAccounts} seats filled. " +
+                                              $"Your new plan only has ({occupiedServiceAccounts}) seats. Remove some users.");
+            }
+        }
+
+        var paymentIntentClientSecret = await _paymentService.AdjustServiceAccountsAsync(organization, plan, additionalServiceAccounts.GetValueOrDefault());
+
+        await _referenceEventService.RaiseEventAsync(
+            new ReferenceEvent(ReferenceEventType.AdjustServiceAccounts, organization, _currentContext)
+            {
+                Id = organization.Id,
+                PlanName = plan.Name,
+                PlanType = plan.Type,
+                ServiceAccounts = newServiceAccountsTotal,
+                PreviousServiceAccounts = organization.SmServiceAccounts
+            });
```

Comment:
Needs to be done last, after all validation has occurred.

Reply:
I am not clear with this comment

Reply:
done the changes. please check